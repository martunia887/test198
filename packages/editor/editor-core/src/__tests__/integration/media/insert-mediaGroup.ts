import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  editable,
  expectMatchDocument,
  comment,
  insertMedia,
} from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';

[comment].forEach(editor => {
  BrowserTestCase(
    `insert-mediaGroup.ts: Inserts a media group on ${editor.name}`,
    { skip: ['edge', 'ie', 'safari'] },
    async (client: any, testName: string) => {
      const page = await goToEditorTestingExample(client);
      await mountEditor(page, {
        appearance: editor.appearance,
        media: {
          allowMediaSingle: false,
          allowMediaGroup: true,
        },
      });

      await page.click(editable);
      await page.type(editable, 'some text');

      // now we can insert media as necessary
      await insertMedia(page);

      expect(await page.isVisible('.wrapper')).toBe(true);

      await expectMatchDocument(page, testName);
    },
  );
});
