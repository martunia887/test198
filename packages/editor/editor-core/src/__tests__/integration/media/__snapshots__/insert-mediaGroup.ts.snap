// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert-mediaGroup.ts: Inserts a media group on comment 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "some text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__fileName": "example.png",
            "collection": "MediaServicesSample",
            "id": "1234-5678-abcd-efgh",
            "type": "file",
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
